src/backends/catalogue_multicore.hpp

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <map>
+#include <string>
+
+#include <ion.hpp>
+#include <mechanism.hpp>
+
+#include "memory_multicore.hpp"
+
+namespace nest {
+namespace mc {
+namespace multicore {
+
+class catalogue : public memory_traits {
```

Comment:
I have moved the catalogue code directly into the `backend` classes. They are so small that it really is a zero-sum game making them generic, and I think it would actually make the code harder to follow (which is important, because every time I refactor, this part of the code breaks). Given that, I prefer to minimise abstraction.